modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -151,11 +152,12 @@ static RemoteInfo buildRemoteInfo(Map<String, Object> source) throws IOException
         String scheme = hostMatcher.group("scheme");
         String host = hostMatcher.group("host");
         int port = Integer.parseInt(hostMatcher.group("port"));
+        Map<String, String> headers = extractStringStringMap(remote, "headers");
         if (false == remote.isEmpty()) {
             throw new IllegalArgumentException(
                     "Unsupported fields in [remote]: [" + Strings.collectionToCommaDelimitedString(remote.keySet()) + "]");
         }
-        return new RemoteInfo(scheme, host, port, queryForRemote(source), username, password);
+        return new RemoteInfo(scheme, host, port, queryForRemote(source), username, password, headers);
```

Comment:
silly question, but what is the usecase for providing headers as part of the remote definition? do you have an example?